packages/frontend/src/utils/blocks.ts

```diff
@@ -0,0 +1,29 @@
+import { providers } from 'ethers'
+
+export async function getRecentBlocks(provider: providers.Provider) {
+  const blockNumber = await provider.getBlockNumber()
+  const nums = [0, 1, 2, 3]
+  const recentBlocks = await Promise.all(
+    nums.map(num => provider.getBlockWithTransactions(blockNumber - num))
+  )
+
+  return recentBlocks
+}
+
+export async function getRecentTransactionsByFromAddress(
+  provider: providers.Provider,
+  fromAddress: string
+) {
+  const blocks = await getRecentBlocks(provider)
+  const txs: providers.TransactionResponse[] = []
+
+  for (const block of blocks) {
+    for (const transaction of block.transactions) {
+      if (transaction.from === fromAddress) {
```

Comment:
Should the addresses here be lowercased to make sure the casing is the same when comparing?

Reply:
it _should_ be ok, since both are checksummed, but i agree we should for best practices